### ISSUE TITLE
[FEATURE] AzureAiSearchEmbeddingStore - add "indexName" to the builder

### ISSUE THREAD

**Issue by vprudnikov** [langchain4j/langchain4j#1062]
**Is your feature request related to a problem? Please describe.**

Currently, it's not possible to override the index name without creating the entire index by yourself. 

**Describe the solution you'd like**

I would like to be able to set the index name using `AzureAiSearchEmbeddingStore.builder()` like 
```
AzureAiSearchEmbeddingStore.builder()
                .apiKey(properties.apiKey())
                .endpoint(properties.endpoint())
                .indexName(myIndexName)
                .dimensions(DIMENSIONS)
                .build();
```

or 

```
AzureAiSearchEmbeddingStore.builder()
                .apiKey(properties.apiKey())
                .endpoint(properties.endpoint())
                .indexName(indexBuilder -> indexBuilder.name(myIndexName))
                .dimensions(DIMENSIONS)
                .build();
```

In other words, it would be great to still rely on the default implementation regarding the index settings however, at the same time, be able to override only the index name.

**Comment by langchain4j** [comment]
cc @jdubois

**Comment by jdubois** [comment]
Thanks, the PR is available at #1084
